src/SignalR/common/Shared/ValueTaskExtensions.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO.Pipelines;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Internal
+{
+    internal static class ValueTaskExtensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Task GetAsTask(this in ValueTask<FlushResult> valueTask)
+        {
+            if (valueTask.IsCompletedSuccessfully)
+            {
+                // Signal consumption to the IValueTaskSource
+                valueTask.GetAwaiter().GetResult();
+                return Task.CompletedTask;
+            }
+            else
+            {
+                return valueTask.AsTask();
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Task GetAsTask(this in ValueTask valueTask)
+        {
+            if (valueTask.IsCompletedSuccessfully)
```

Comment:
AsTask is already optimized for ValueTask => https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/Threading/Tasks/ValueTask.cs#L170.

Reply:
Why was this marked as resolved?

Reply:
Because GitHub has bugs and this comment is for a previous commit on code that no longer exists